src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -137,215 +176,135 @@ public void AnalyzeCodeBlockEnd(CodeBlockAnalysisContext context)
+                        AnalyzeInvocation(context);
+                        break;
+                }
             }
 
-            private void AnalyzeNodeForDtdProcessingOverloads(SyntaxNodeAnalysisContext context)
+            private void AnalyzeInvocation(OperationAnalysisContext context)
             {
-                SyntaxNode node = context.Node;
-                SemanticModel model = context.SemanticModel;
+                IInvocationExpression invoke = context.Operation as IInvocationExpression;
 
-                IMethodSymbol method = _syntaxNodeHelper.GetCalleeMethodSymbol(node, model);
-                if (method == null)
+                if(invoke == null)
                 {
                     return;
                 }
 
-                CompilationSecurityTypes xmlTypes = _xmlTypes;
-                if (method.MatchMethodDerivedByName(xmlTypes.XmlDocument, SecurityMemberNames.Load) ||                                    //FxCop CA3056
-                    method.MatchMethodDerivedByName(xmlTypes.XmlDocument, SecurityMemberNames.LoadXml) ||                                 //FxCop CA3057
-                    method.MatchMethodDerivedByName(xmlTypes.XPathDocument, WellKnownMemberNames.InstanceConstructorName) ||         //FxCop CA3059
-                    method.MatchMethodDerivedByName(xmlTypes.XmlSchema, SecurityMemberNames.Read) ||                                      //FxCop CA3060
-                    method.MatchMethodDerivedByName(xmlTypes.DataSet, SecurityMemberNames.ReadXml) ||                                     //FxCop CA3063
-                    method.MatchMethodDerivedByName(xmlTypes.DataSet, SecurityMemberNames.ReadXmlSchema) ||                               //FxCop CA3064
-                    method.MatchMethodDerivedByName(xmlTypes.XmlSerializer, SecurityMemberNames.Deserialize) ||                           //FxCop CA3070
-                    method.MatchMethodDerivedByName(xmlTypes.DataTable, SecurityMemberNames.ReadXml) ||                                   //FxCop CA3071
-                    method.MatchMethodDerivedByName(xmlTypes.DataTable, SecurityMemberNames.ReadXmlSchema))                               //FxCop CA3072
+                IMethodSymbol method = invoke.TargetMethod;
+
+                if(method == null)
+                {
+                    return;
+                }
+
+                AnalyzeMethodOverloads(context, method, invoke.ArgumentsInParameterOrder, invoke.Syntax.GetLocation());
+            }
+
+            private void AnalyzeMethodOverloads(OperationAnalysisContext context, IMethodSymbol method, ImmutableArray<IArgument> arguments, Location location)
+            {
+                SemanticModel model = context.Compilation.GetSemanticModel(context.Operation.Syntax.SyntaxTree);
+
+                if (method.MatchMethodDerivedByName(_xmlTypes.XmlDocument, SecurityMemberNames.Load) ||                                    //FxCop CA3056
```

Comment:
I looked at all the callers of `MatchMethodDerivedByName` and it is always a IMethodSymbol. So if you change the method signature to `public static bool MatchMethodDerivedByName(this IMethodSymbol member, INamedTypeSymbol type, string name)` instead of `public static bool MatchMethodDerivedByName(this ISymbol member, INamedTypeSymbol type, string name)` then that method does not need the `member.Kind == SymbolKind.Method`